sunpy/net/dataretriever/sources/eve.py

```diff
@@ -64,8 +66,8 @@ def _get_url_for_timerange(self, timerange, **kwargs):
         # If start of time range is before 00:00, converted to such, so
         # files of the requested time ranger are included.
         # This is done because the archive contains daily files.
-        if timerange.start.time() != datetime.time(0, 0):
-            timerange = TimeRange('{:%Y-%m-%d}'.format(timerange.start), timerange.end)
+        if timerange.start.datetime.time() != datetime.time(0, 0):
```

Comment:
Find a better way to do this.

Is 
```python
timerange.start.strftime('%M-%S') == '00-00'
``
acceptable? I find it very hacky.

Reply:
What's wrong with this as you have implemented? It seems ok to me.